autonomy_core/map_plan/mapper/src/local_global_mapper.cpp

```diff
@@ -288,9 +288,38 @@ void LocalGlobalMapperNode::processCloud(
                                             T_odom_lidar.translation().y(),
                                             T_odom_lidar.translation().z());
 
+  const Eigen::Vector3d global_center(global_map_cx_,
+                                      global_map_cy_,
+                                      global_map_cz_);
+
   ros::Time t0;
   cpu_times tc;
 
+  // Reset global and storage maps if the robot is far from the map centers
+  const auto diff = lidar_position_odom - global_center;
+  const auto distance = diff.norm();
+  const auto threshold = std::min(global_map_dim_d_x_, global_map_dim_d_x_) / 40;
+  ROS_INFO("[Mapper]: distance is %.2lf, threshold is %.2lf", distance, threshold);
+  if (distance > threshold) {
+    ROS_INFO("[Mapper]: resetting global and storage map");
```

Comment:
Please add a TODO here stating that we need to publish a stopping policy state trigger